src/IK/include/BipedalLocomotion/IK/InverseKinematics.h

```diff
@@ -0,0 +1,96 @@
+/**
+ * @file InverseKinematics.h
+ * @authors Giulio Romualdi
+ * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_IK_INVERSE_KINEMATICS_H
+#define BIPEDAL_LOCOMOTION_IK_INVERSE_KINEMATICS_H
+
+#include <memory>
+#include <string>
+#include <optional>
+
+#include <Eigen/Dense>
+#include <manif/SE3.h>
+
+#include <BipedalLocomotion/ParametersHandler/IParametersHandler.h>
+#include <BipedalLocomotion/System/Advanceable.h>
+#include <BipedalLocomotion/IK/Task.h>
+
+namespace BipedalLocomotion
+{
+
+namespace IK
+{
+
+/**
+ * State of the InverseKinematics
+ */
+struct IKState
+{
+    Eigen::VectorXd jointVelocity; /**< Joints velocity in rad per seconds */
+    manif::SE3d::Tangent baseVelocity; /**< Mixed spatial velocity of the base */
+};
+
+/**
+ * InverseKinematics implements the interface for the inverse kinematics. Please inherits this class
+ * if you want to implement your custom IK.
+ */
+class InverseKinematics : public BipedalLocomotion::System::Advanceable<IKState>
```

Comment:
This seems something that gives in output the robot velocity, so could it make sense to call it `VelocityInverseKinematics` , to distinguish it from the position-based kinematics?